Bottom: e5a2323699818c850ce90a611481d160010b7fcb
Top:    b0f8531f5801564754e4b85174407639f873fd2f
Author: Andrew Cooper <andrew.cooper@novatechweb.com>
Date:   2018-08-29 16:23:24 -0500

Refresh of fixup-asset-host-3

---
diff --git a/roles/buildbot-master/files/ptxdist.py b/roles/buildbot-master/files/ptxdist.py
index c6e148cf..c6e563b8 100644
--- a/roles/buildbot-master/files/ptxdist.py
+++ b/roles/buildbot-master/files/ptxdist.py
@@ -194,6 +194,16 @@ class PTXDistBuild(steps.ShellSequence):
                     "-f", util.Interpolate("%(prop:artifact_dest)s/%(prop:ipkg_artifact)s"),
                     util.Property("project")
                 ]),
+
+            util.ShellArg(
+                logfile="upload",
+                haltOnFailure=False,
+                flunkOnFailure=True,
+                command=[
+                    'curl', '--netrc', '--verbose',
+                    '--upload-file', util.Property('ipkg_artifact'),
+                    '--url', util.Property('ipkg_url')
+                ]),
         ]
 
 
@@ -220,8 +230,10 @@ class PTXDistImages(steps.ShellSequence):
             util.ShellArg(
                 logfile="ptxdist images",
                 haltOnFailure=True,
-                command=["ptxdist", "images"],
-            ),
+                command=[
+                    "ptxdist",
+                    "images"
+                ]),
 
             util.ShellArg(
                 logfile="archive",
@@ -231,8 +243,17 @@ class PTXDistImages(steps.ShellSequence):
                     "-C", util.Property("image_root"),
                     "-f", util.Interpolate("%(prop:artifact_dest)s/%(prop:image_artifact)s"),
                     "."
-                ]
-            ),
+                ]),
+
+            util.ShellArg(
+                logfile="upload",
+                haltOnFailure=False,
+                flunkOnFailure=True,
+                command=[
+                    'curl', '--netrc', '--verbose',
+                    '--upload-file', util.Property('image_artifact'),
+                    '--url', util.Property('image_url')
+                ]),
         ]
 
 
@@ -287,6 +308,37 @@ def ComputeBuildProperties(props):
         props.getProperty('platform')
     )
 
+    if props.getProperty('release'):
+        urlpath = os.path.join(
+            'ptxdist',
+            'release',
+            version
+        )
+    elif props.getProperty('scheduler') == "ptxdist-nightly":
+        urlpath = os.path.join(
+            'ptxdist',
+            'nightly',
+            version
+        )
+    else:
+        urlpath = os.path.join(
+            'ptxdist',
+            'beta',
+            version
+        )
+
+    newprops['ipkg_url'] = os.path.join(
+        ASSET_HOST,
+        urlpath,
+        ipkg_artifact,
+    )
+
+    newprops['image_url'] = os.path.join(
+        ASSET_HOST,
+        urlpath,
+        image_artifact,
+    )
+
     return newprops
